### DISCUSSION TITLE
[Bug report] Element is missing end tag.

### DISCUSSION THREAD

**Discussion by web1992** [vuepress/docs#14] [Q&A] [answered]
### Description

```md
# Guide

This is guide page.

<div>

```

当我使用 div 标签，没有闭合标签结尾的时候，就出现下面的错误。

![image](https://github.com/vuepress/docs/assets/6828647/44553742-2359-4c9f-ac34-6f9c9240852c)

这种情况怎么解决？比如Java 中的泛型 ` List<String>` 也会遇到这样问题。
问题的原因是 md 文件被解析成了vue 文件，而vue 文件中 template是不能存在没有闭合的 html 标签的。

目前可以使用下面的方式解决。有其他更好的方式吗?
```md
`<div>`
```

**Comment by Mister-Hope** [comment, marked as answer]
why do you think it's an issue?

Write some standard markdown plz. 

List\<string>

```md
List\<string>
```

**Reply by web1992** [reply]
单纯的Markdown 这个写法`List<string>`其实是没有问题的，依然可以解析，不是markdwon 的问题，而是Vue 的问题。

**Reply by Mister-Hope** [reply]
? Learn some markdown basic first before pushing any point to others.

```md
- <string>
- \<string>
- <string\>
```

Rendered as :

- <string>
- \<string>
- <string\>

Then why saying "单纯的Markdown 这个写法List<string>其实是没有问题的"? Because markdown doesn't throw? That's right, you can push shit to html like this:

```html
<h1>Shit
<p>This is shit
```

The browser is designed to render as possible, so it will try to render the content by trying to closing any tag before others, and completing the missing h1 and p for you so it becomes:

```html
<html>
<body>
<h1>Shit</h1>
<p>This is shit</p>
</body>
</html>
```

But if you add an unclosing body, you will get an empty result (only 1 body is allowed):

```html
<body>
<h1>Shit
<p>This is shit
```

becomes:

```html
<html>
<body></body>
<h1>Shit</h1>
<p>This is shit</p>
</html>
```

So even browsers can not render correctly without closing tag. That's why mostfront-end tools  requires you to provide standard html.


Vue does, any unknown tag will be treated as Vue components, and refuse to parse any unstandard template. 

When you provide an unknown tag `<string>` or any unclosing tag that is not empty, it throws, and this is expected under design. It this clear enough for you?

Learn basics before arguing anything with others. It's not my fault that you are not familiar with something, but it's your fault to claiming something wrong! Open source authors has no responsibility teaching you basic like this and you are wasting my time.
